packages/just-repo-utils/src/internal/packageInfoHelpers.ts

```diff
@@ -0,0 +1,86 @@
+import { PackageJson, PackageJsonLoader, RushProject } from '../interfaces/configTypes';
+import { PackageEntries, PackageEntry, PackageInfo } from '../interfaces/packageInfoTypes';
+import { getConfigLoader, readPackageJson } from '../readConfigs';
+import { normalizePath } from '../normalizePath';
+import path from 'path';
+import glob from 'glob';
+
+function buildPackageInfo(root: string, pkgPath: string, pkgJsonName?: string): PackageEntries {
+  const pkgJsonPath = pkgJsonName ? path.join(root, pkgPath, pkgJsonName) : path.join(root, pkgPath);
+  const getConfig = getConfigLoader<PackageJson>(pkgJsonPath) as PackageJsonLoader;
+  return {
+    [getConfig().name]: {
+      path: normalizePath(path.dirname(pkgJsonPath)),
+      getConfig,
+      dependencies: {}
+    }
+  };
+}
+
+function buildPackageInfoForGlob(root: string, pkgGlob: string): PackageEntries {
+  const matchPattern = pkgGlob + '/package.json';
+  const globOptions = { cwd: root, ignore: '**/node_modules/**' };
+  return Object.assign({}, ...glob.sync(matchPattern, globOptions).map(subPath => buildPackageInfo(root, subPath)));
+}
+
+function addPackageDependencyBranch(repoInfo: PackageEntries, pkg: PackageEntry, key: string): void {
+  const config = pkg.getConfig();
+  const section = config[key];
+  const dependencies = pkg.dependencies;
+  if (section) {
+    Object.keys(section).forEach(dependency => {
+      if (repoInfo[dependency] && !dependencies[dependency]) {
+        dependencies[dependency] = repoInfo[dependency];
+      }
+    });
+  }
+}
+
+function addPackageDependencies(info: PackageEntries): void {
+  Object.keys(info).forEach(pkg => {
+    addPackageDependencyBranch(info, info[pkg], 'dependencies');
+    addPackageDependencyBranch(info, info[pkg], 'devDependencies');
+  });
+}
+
+export function buildPackageInfoFromGlobs(root: string, globs: string[]): PackageEntries {
+  const results: PackageEntries = Object.assign({}, ...globs.map(glob => buildPackageInfoForGlob(root, glob)));
+  addPackageDependencies(results);
+  return results;
+}
+
+export function buildPackageInfoFromRushProjects(root: string, projects: RushProject[]): PackageEntries {
+  const results: PackageEntries = Object.assign(
+    {},
+    ...projects.map(project => buildPackageInfo(root, project.projectFolder, 'package.json'))
+  );
+  addPackageDependencies(results);
+  return results;
+}
+
+function addRecursiveDependencies(collector: PackageEntries, entry: PackageEntry): void {
+  const dependencies = entry.dependencies;
```

Comment:
There should be compliments of this for devDependencies maybe even peerDependencies depending on your needs.

Reply:
Hmmm...right now this rolls both dependencies and devDependencies together as when building an in-repo graph that is typically what is wanted.  I can split them while keeping the default behavior the same.

Reply:
Published an update here.  I looked at splitting them but the issue is that from a yarn.lock perspective (which is what the caching is based on) there is no difference between dependencies and devDependencies.  

Instead what I did is add an option for filtering the results.  If this is specified the package.json files will be loaded when needed to see whether the dependency is dev or not.  I also added some snapshot tests for traversing these dependencies.